include/qore/intern/qore_number_private.h

```diff
@@ -443,6 +443,44 @@ struct qore_number_private : public qore_number_private_intern {
       return p;
    }
 
+    // for round functions: round(), ceil(), floor()
+    DLLLOCAL qore_number_private* doRoundNR(q_mpfr_unary_nr_func_t func, int prec = 0, ExceptionSink* xsink = NULL) const {
+        qore_number_private* p0 = new qore_number_private(*this);
```

Comment:
please use std::unique_ptr<> for all possibly temporary values instead of manually deleting them - it's one of our C++ best practices